### PR TITLE
Run smoke e2e test during PR build

### LINKED ISSUE #902: Run basic E2E tests on PR builds
We should run basic E2E tests (a subset of the entire list) on PR builds in CI that cover creating:

* ES cluster
* Kibana
* APM

No need to test complex scenarios (mutations, version upgrades, failure scenarios) that are covered by other e2e tests.

The main goal here is to make sure we at least go through the `Reconcile()` and nothing is obviously broken in there.

Should we try to reuse an existing GKE cluster to avoid waiting to spawn a new one on each PR?

### PR DESCRIPTION
**Pull request by artemnikitin**
Need to merge https://github.com/elastic/cloud-on-k8s/pull/900 first, otherwise it will fail at https://github.com/elastic/cloud-on-k8s/blob/master/operators/test/e2e/stack/checks_es.go#L79
close #902

### CODE REVIEW

**Comment by artemnikitin** [comment]
> IIUC we'll create a new GKE cluster for each PR. Let's see how it goes and maybe we can optimize with reusing a dedicated cluster later.

Yes, I decided re-use existed stuff to submit PR quickly and we can optimize lately if there will be issues with speed. But honestly I think we are ok. Smoke test itself runs 157 sec (2.5 min)

File: operators/test/e2e/smoke_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+	namespacedSampleStack := sampleStack.WithNamespace(helpers.DefaultNamespace)
+	namespacedSampleApm := sampleApm.WithNamespace(helpers.DefaultNamespace)
+
+	k := helpers.NewK8sClientOrFatal()
+	helpers.TestStepList{}.
+		WithSteps(stack.InitTestSteps(namespacedSampleStack, k)...).
+		WithSteps(apm.InitTestSteps(namespacedSampleApm, k)...).
+		WithSteps(stack.CreationTestSteps(namespacedSampleStack, k)...).
+		WithSteps(apm.CreationTestSteps(namespacedSampleApm, namespacedSampleStack.Elasticsearch, k)...).
```
Comment (thbkrkr) [inline operators/test/e2e/smoke_test.go]: To be sure: we don't use the `DeletionTestSteps` to gain a few seconds in the total running time?

Comment (artemnikitin) [reply operators/test/e2e/smoke_test.go]: We don't use `DeletionTestSteps`. I can add them if you feel it will make tests better

Comment (artemnikitin) [reply operators/test/e2e/smoke_test.go]: I checked and with `DeletionTestSteps` test runs ~ 1 min more (205 sec vs 157 sec). Main question will it be helpful? That do you think @thbkrkr ?

**Pull request merged**